script/MoveMediaManager.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Integrates move capability into the media manager
+ * Based on the implementation in diagrams plugin
+ */
+class MoveMediaManager {
+
+    constructor() {
+        const filePanel = document.querySelector('#mediamanager__page .panel.file');
+        if (filePanel) {
+            const observer = new MutationObserver(this.#addEditButton.bind(this));
+            observer.observe(filePanel, {childList: true, subtree: true});
+        }
+    }
+
+    /**
+     * Observer callback to add the edit button in the detail panel of the media manager
+     *
+     * @param mutationsList
+     * @param observer
+     */
+    async #addEditButton(mutationsList, observer) {
+        for (let mutation of mutationsList) {
+            // div.file has been filled with new content?
+            if (mutation.type !== 'childList') continue;
+
+            // check that the file panel contains a link to a file
+            if (mutation.target.classList.contains('file') === false) continue;
+            const link = mutation.target.querySelector('a.select.mediafile');
+            if (!link) continue;
+
+            const uri = link.getAttribute('href');
+
+            const [ , paramString ] = uri.split( '?' );
+            const searchParams = new URLSearchParams(paramString);
+            const src = searchParams.get('media');
+
+            const actionList = mutation.target.querySelector('ul.actions');
+            if (actionList.querySelector('button.move-btn')) continue; // already added
+
+
+            const moveButton = document.createElement('button');
```

Comment:
Wouldn't it be better to first check if the user can at least delete the media file at the source location? I think we shouldn't display a move button to user who can't at least delete the media file as otherwise the move operation will always fail. I think it should be possible to do this by checking if there is a delete button in the actions.